.github/workflows/release_ci.yml

```diff
@@ -12,12 +12,15 @@ on:
   schedule:
     - cron: '0 1 * * 4' # this means every Thursday @1am UTC
 
+concurrency: release
+
 jobs:
   release_wrkflw:
     name: Do release
-    uses: PaloAltoNetworks/terraform-modules-vmseries-ci-workflows/.github/workflows/release_ci.yml@v1.2.0
+    uses: PaloAltoNetworks/terraform-modules-vmseries-ci-workflows/.github/workflows/release_ci.yml@makefile-with-terratest
     secrets: inherit
     with:
       cloud: azure
-      max_parallel: 10
-      do_apply: true
+      validate_max_parallel: 20
+      test_max_parallel: 5
+      terratest_action: Idempotence # keep in mind that this has to start with capital letter
```

Comment:
What is the reason to have capital letters for Terratest actions ?

Reply:
the action is being passed to `go test -run HERE` directly (passing through sub-workflows, action, Makefile). You can pass all-lowercase, like this:
```bash
make examples/common_vmseries ACTION=validate
::group::DOWNLOADING GO DEPENDENCIES
::endgroup::
::group::ACTION >>validate<<
testing: warning: no tests to run
PASS
ok  	github.com/PaloAltoNetworks/terraform-azure-vmseries-modules/examples/common_vmseries	1.694s
::endgroup::
```

Something is being run, but I do not know what TBH. 
When you run it with 1st letter capital, it matches the test name. You can see the output is different:

```bash
make examples/common_vmseries ACTION=Validate
::group::DOWNLOADING GO DEPENDENCIES
::endgroup::
::group::ACTION >>Validate<<
TestValidate 2023-08-29T12:00:56+02:00 retry.go:91: terraform [init -upgrade=true]
TestValidate 2023-08-29T12:00:56+02:00 command.go:100: Running command terraform with args [init -upgrade=true]
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: Upgrading modules...
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: - ai in ../../modules/application_insights
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: - appgw in ../../modules/appgw
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: - bootstrap in ../../modules/bootstrap
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: - bootstrap_share in ../../modules/bootstrap
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: - load_balancer in ../../modules/loadbalancer
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: - natgw in ../../modules/natgw
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: - vmseries in ../../modules/vmseries
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: - vnet in ../../modules/vnet
TestValidate 2023-08-29T12:00:56+02:00 command.go:185:
TestValidate 2023-08-29T12:00:56+02:00 command.go:185: Initializing the backend...
TestValidate 2023-08-29T12:00:57+02:00 command.go:185:
TestValidate 2023-08-29T12:00:57+02:00 command.go:185: Initializing provider plugins...
TestValidate 2023-08-29T12:00:57+02:00 command.go:185: - Finding hashicorp/azurerm versions matching "~> 3.25"...
TestValidate 2023-08-29T12:00:57+02:00 command.go:185: - Finding hashicorp/random versions matching "~> 3.1"...
TestValidate 2023-08-29T12:00:57+02:00 command.go:185: - Finding latest version of hashicorp/http...
TestValidate 2023-08-29T12:00:58+02:00 command.go:185: - Finding latest version of hashicorp/local...
TestValidate 2023-08-29T12:00:58+02:00 command.go:185: - Using previously-installed hashicorp/random v3.5.1
TestValidate 2023-08-29T12:00:58+02:00 command.go:185: - Using previously-installed hashicorp/http v3.4.0
TestValidate 2023-08-29T12:00:58+02:00 command.go:185: - Using previously-installed hashicorp/local v2.4.0
TestValidate 2023-08-29T12:00:58+02:00 command.go:185: - Installing hashicorp/azurerm v3.71.0...
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: - Installed hashicorp/azurerm v3.71.0 (signed by HashiCorp)
TestValidate 2023-08-29T12:01:06+02:00 command.go:185:
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: Terraform has made some changes to the provider dependency selections recorded
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: in the .terraform.lock.hcl file. Review those changes and commit them to your
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: version control system if they represent changes you intended to make.
TestValidate 2023-08-29T12:01:06+02:00 command.go:185:
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: Terraform has been successfully initialized!
TestValidate 2023-08-29T12:01:06+02:00 command.go:185:
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: You may now begin working with Terraform. Try running "terraform plan" to see
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: any changes that are required for your infrastructure. All Terraform commands
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: should now work.
TestValidate 2023-08-29T12:01:06+02:00 command.go:185:
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: If you ever set or change modules or backend configuration for Terraform,
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: rerun this command to reinitialize your working directory. If you forget, other
TestValidate 2023-08-29T12:01:06+02:00 command.go:185: commands will detect it and remind you to do so if necessary.
TestValidate 2023-08-29T12:01:06+02:00 retry.go:91: terraform [validate]
TestValidate 2023-08-29T12:01:06+02:00 command.go:100: Running command terraform with args [validate]
TestValidate 2023-08-29T12:01:19+02:00 command.go:185: Success! The configuration is valid.
TestValidate 2023-08-29T12:01:19+02:00 command.go:185:
PASS
ok  	github.com/PaloAltoNetworks/terraform-azure-vmseries-modules/examples/common_vmseries	23.190s
::endgroup::
```

Reply:
It's clear, thank you for explanation 👍